src/server/src/main/java/io/cassandrareaper/AppContext.java

```diff
@@ -49,6 +51,7 @@
   public JmxConnectionFactory jmxConnectionFactory;
   public ReaperApplicationConfiguration config;
   public MetricRegistry metricRegistry = new MetricRegistry();
+  public Set<String> accessibleDatacenters = Sets.newHashSet();
```

Comment:
@alex i believe we can remove this.

it seems only to be used by tests. 
and those assertions in `ClusterFacadeTest` can be done instead against `context.jmxConnectionFactory.java.getHostConnectionCounters()`

Reply:
it's used in ClusterFacade actually : https://github.com/thelastpickle/cassandra-reaper/pull/619/files#diff-3475cbd1fe679c34b1b40e562c7db6daR446

And it's a good first step to handle operations efficiently when it comes to use JMX or the backend. 
Please, let's keep it in this PR.

Reply:
> And it's a good first step to handle operations efficiently when it comes to use JMX or the backend.

Oh I missed that!
I agree that the code functionality in `ClusterFacade` should stay. It can be done though without adding the state to the content.

Can you add this method to `ClusterFacade`
```
@VisibleForTesting
boolean isDatacenterAccessible(String datacenter) {
  for (String host : context.jmxConnectionFactory.java.getHostConnectionCounters().getSuccessfullyConnectedHosts()) {
    if (EndpointSnitchInfoProxy.create(proxy).getDataCenter(host).equals(datacenter)) {
      return true;
    }
  }
  return false;
}
```

And in `HostConnectionCounters` the method `.getSuccessfullyConnectedHosts()` needs to be added which returns `Set<String>` for all the hosts that have one or more successful connections.

Reply:
My concern with your solution is that we'll have to issue a JMX call just to check if the node is accessible through JMX.
I could move `accessibleDatacenters` to `JmxConnectionFactory` instead if you don't like having it in `AppContext`.
Regarding https://github.com/thelastpickle/cassandra-reaper/pull/619/files#diff-739e34b5f2ca85ff72da4077afc1ea01R230 , it's storing the datacenter of the host we're connected to, which is the desired behavior (if we detect that the connection is possible to a host then we add its dc to the list of accessible ones). The call just avoids doing this at each successful connection and does it only if we previously failed or never connected to this node.

Reply:
> Regarding https://github.com/thelastpickle/cassandra-reaper/pull/619/files#diff-739e34b5f2ca85ff72da4077afc1ea01R230 , it's storing the datacenter of the host we're connected to

too true, sorry about that.

> My concern with your solution is that we'll have to issue a JMX call just to check if the node is accessible through JMX.
> I could move `accessibleDatacenters` to `JmxConnectionFactory` instead if you don't like having it in `AppContext`.

The jmx call can be also avoided with caching its values. It would make sense to have a guava cache inside `EndpointSnitchInfoProxy.getDataCenter(host)` as it's a low-cardinality lookup.
But either way is fine for me, what's important for me is to go from the "global constant" to encapsulation.

Reply:
> It would make sense to have a guava cache inside `EndpointSnitchInfoProxy.getDataCenter(host)` as it's a static and low-cardinality lookup.

In fact adding this type of logic was one of the reasons to create the separate AbcProxy classes above JmxProxy.

Reply:
New commit https://github.com/thelastpickle/cassandra-reaper/commit/6d9bd1b73c6e4dce6d5bdb381c665c608b65f7aa
 that fixes the unit tests (`jmxProxyImpl.getUntranslatedHost()` needed mocking)